### PR TITLE
Bug 1782404: Remove deprecated check in update

### PR DESCRIPTION
**Pull request by alexander-demichev**
Remove unnecessary check that blocks users from doing manual changes

### CODE REVIEW

File: pkg/cloud/azure/actuators/machine/reconciler.go [hunk 1]
```diff
@@ -136,13 +135,6 @@ func (s *Reconciler) Update(ctx context.Context) error {
 		return errors.New("returned incorrect vm interface")
 	}
 
-	// We can now compare the various Azure state to the state we were passed.
-	// We will check immutable state first, in order to fail quickly before
-	// moving on to state that we can mutate.
-	if isMachineOutdated(s.scope.MachineConfig, converters.SDKToVM(vm)) {
```
Comment (enxebre) [inline pkg/cloud/azure/actuators/machine/reconciler.go]: is `SDKToVM` used somewhere else? if not can we please drop it?

Comment (alexander-demichev) [reply pkg/cloud/azure/actuators/machine/reconciler.go]: you're right

**Pull request merged**